tests/targets/_internal/test_board_database.py

```diff
@@ -85,11 +85,19 @@ def test_no_auth_header_set_with_empty_token_var(self, requests):
         board_database._get_request()
         requests.get.assert_called_once_with(board_database._BOARD_API, headers=None)
 
+    @mock.patch("mbed_tools.targets._internal.board_database.logger.warning", autospec=True)
+    @mock.patch("mbed_tools.targets._internal.board_database.requests.get")
+    def test_logs_no_warning_on_success(self, get, logger_warning):
+        board_database._get_request()
+        logger_warning.assert_not_called()
+
+    @mock.patch("mbed_tools.targets._internal.board_database.logger.warning", autospec=True)
     @mock.patch("mbed_tools.targets._internal.board_database.requests.get")
-    def test_raises_tools_error_on_connection_error(self, get):
+    def test_raises_tools_error_on_connection_error(self, get, logger_warning):
         get.side_effect = board_database.requests.exceptions.ConnectionError
         with self.assertRaises(board_database.BoardAPIError):
             board_database._get_request()
+        logger_warning.assert_called_once()
```

Comment:
This'll work, but there is a [pytest `caplog` builtin fixture](https://docs.pytest.org/en/stable/logging.html#caplog-fixture) you could use for more elegance.